src/abstract-command.js

```diff
@@ -297,6 +303,111 @@ class AbstractCommand {
     this.logger.warn('THUB_TOKEN is not provided.');
     return Promise.resolve();
   }
+
+
+  /**
+   *   ******************* NEW *******************
+   */
+
+  /**
+   * @param {Object} fullConfig
+   * @returns {Promise}
+   */
+  checkProjectDuplicateComponents(fullConfig) {
+    const result = {};
+
+    Object.keys(fullConfig).forEach(hash => {
+      const { name, root } = fullConfig[hash];
+
+      if (result.hasOwnProperty(name)) {
+        result[name].push(root);
+      } else {
+        result[name] = [root];
+      }
+    });
+
+    const duplicates = Object.keys(result).filter(it => result[it].length > 1);
+
+    if (duplicates.length) {
+      const messages = duplicates.map(it => `component '${it}' ` +
+        `has duplicates in '${result[it].join(`' ,'`)}' directories`);
+
+      throw new ListException(messages, {
+        header: 'Some components have duplicates in project:',
+        style: ListException.NUMBER
+      });
+    }
+
+    return Promise.resolve();
+  }
+
+  /**
+   * @return {Promise}
+   */
+  checkProjectDataMissing() {
```

Comment:
check if this function is used in `DistributedCommand` class

Reply:
transferred to `DistributedCommand `